harness/determined/keras/_callback.py

```diff
@@ -0,0 +1,446 @@
+        return bool(self.params["verbose"] != 0)
+
+    def _implements_predict_batch_hooks(self) -> bool:
+        # Tell keras that we don't actually want any on_predict_batch_end calls.
+        return False
+
+    def _print_progress(
+        self, logs: Optional[Dict[str, Any]], training: bool, batches: int, total: Optional[int]
+    ) -> None:
+        # Only report progress if we have the target total.
+        if total is None:
+            return
+
+        # Don't report more often than 10% increments.
+        percent_10 = int((batches / total) * 10) * 10
+        if percent_10 <= self._percent_reported:
+            return
+
+        # When you do report, report to 1% accuracy.
+        percent = int((batches / total) * 100)
+        self._percent_reported = percent
+
+        if training:
+            report = (
+                f"total batches trained: {self._steps_completed}, "
+                f"epoch {percent}% complete ({batches}/{total})"
+            )
+        else:
+            report = (
+                f"validation after batch: {self._steps_completed}, "
+                f"validation {percent}% complete ({batches}/{total})"
+            )
+        if logs is not None:
+            metrics = {k: v for k, v in logs.items() if k not in ("batch", "size")}
+            report += f": {metrics}"
+
+        print(report)
+
+    def on_train_begin(self, logs: Optional[Dict[str, Any]]) -> None:
+        # Load initial state.  Note that we might set model._training_state, which is how we
+        # override the initial_epoch provided to model.fit(), but this callback occurs just before
+        # model.fit() trys to read model._training_state.
+        self._ckpt_context = self._load(self._checkpoint)
+
+    def on_epoch_begin(self, epoch: int, logs: Optional[Dict[str, Any]]) -> None:
+        if not self._is_chief:
+            return
+
+        # Set status.
+        self._core.train.set_status("training")
+
+        # Print progress.
+        if self._is_chief and self.params["verbose"] != 0:
+            self._training_batches = 0
+            self._percent_reported = -1
+            self._print_progress(logs=None, training=True, batches=0, total=self._training_length)
+
+    def on_train_batch_end(self, batch: int, logs: Optional[Dict[str, Any]]) -> None:
+        self._steps_completed += 1
+
+        # Delete the initial checkpoint files, if we haven't already.
+        if self._ckpt_context:
+            self._ckpt_context.close()
+            self._ckpt_context = None
+
+        if not self._is_chief:
+            return
+
+        assert logs
+
+        # Report metrics.
+        if (
+            isinstance(self._report_period, int)
+            and self._steps_completed % self._report_period == 0
+        ):
+            self._core.train.report_metrics("training", self._steps_completed, logs)
+
+        # Print progress.
+        if self.params["verbose"] != 0:
+            self._training_batches += 1
```

Comment:
why don't we use the batch param that keras provides?

Reply:
hm... I think we could, but does the `on_epoch_end` have the `batch` in the `logs`?  We need to capture it for subsequent epochs.

Reply:
yeah idk, seems risky to rely on logs